src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/service/IdamService.java

```diff
@@ -0,0 +1,23 @@
+package uk.gov.hmcts.reform.iacasedocumentsapi.domain.service;
+
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.iacasedocumentsapi.infrastructure.clients.IdamApi;
+import uk.gov.hmcts.reform.iacasedocumentsapi.infrastructure.clients.model.idam.UserInfo;
+
+@Component
+public class IdamService {
+
+    private final IdamApi idamApi;
+
+    public IdamService(
+        IdamApi idamApi
+    ) {
+        this.idamApi = idamApi;
+    }
+
+    @Cacheable(value = "userInfoCache")
```

Comment:
This, on its own, is not wise in this context.

Cached values by default remain stored as long as the component lives. This means that potentially we'll be storing authentication information on somebody for an unlimited amount of time.

As among the information we store there is also information about the roles (which may change over time), it would be wise to add an eviction strategy. Check the `@CacheEvict` and `@Scheduled` annotations in Spring.

I believe you'll need to add a method like the following, at a minimum:
```
@CacheEvict(value = "userInfoCache")
@Scheduled(fixedRate = 60, timeUnit = TimeUnit.MINUTES)
public void evictingUserInfoCacheEntry(String accessToken) {
    logger.debug("Evicting entry from userInfo cache");
}
```

Make sure other repositories with the same type of change also have a similar eviction strategy for their cache.